### PR TITLE
HOPSWORKS-772

### PR DESCRIPTION
**Pull request by tkakantousis**
During an upgrade, we need to remove the `tensorflow_demo` directory before putting the `demo.tar.gz` contents in there.

### CODE REVIEW

**Comment by tkakantousis** [comment]
This is intended as a 0.6 bugfix. From 0.7 data under `/user/hdfs/tensorflow_demo` will be versioned.

**Comment by robzor92** [comment]
Did we agree on removing it and not versioning?

File: recipes/default.rb [hunk 1]
```diff
@@ -56,6 +56,10 @@
     not_if { ::File.exists?("#{node['tensorflow']['examples_version']}/.installed") }
   end
 
+  hops_hdfs_directory "/user/#{node['hops']['hdfs']['user']}/#{node['tensorflow']['hopstfdemo_dir']}" do
+    action :rm_as_superuser
+  end
```
Comment (SirOibaf) [inline recipes/default.rb]: Why not using the `:replace_as_superuser`?

Comment (kouzant) [reply recipes/default.rb]: I agree with @SirOibaf

Comment (tkakantousis) [reply recipes/default.rb]: I replaced it with `replace_as_superuser`

**Pull request merged**